src/vs/workbench/browser/window.ts

```diff
@@ -231,13 +233,13 @@ export class BrowserWindow extends Disposable {
 							);
 						}
 
-						await this.dialogService.prompt({
+						await this.hostService.withExpectedShutdown(() => this.dialogService.prompt({
```

Comment:
A comment above to explain why we run the dialog in the new method would help understand it better.